In [ ]:
from google.colab import files
uploaded = files.upload()

import pandas as pd
data = pd.read_csv('krebs.csv')


Saving krebs.csv to krebs.csv


In [ ]:
!pip install autokeras
!pip install tensorflow


     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 148.6/148.6 kB 2.9 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 128.9/128.9 kB 13.2 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 584.8/584.8 kB 24.1 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 950.8/950.8 kB 43.5 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.2/5.2 MB 68.4 MB/s eta 0:00:00


In [3]:
from sklearn.model_selection import train_test_split
from autokeras import StructuredDataClassifier

# Подготовка данных
# Предположим, что последний столбец - это целевая переменная
X = data.iloc[:, :-1]
y = data.iloc[:, -1]

# Разделение данных на обучающую, валидационную и тестовую выборки
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Создание и обучение модели AutoKeras
model = StructuredDataClassifier(max_trials=10) # Вы можете изменить количество испытаний
model.fit(X_train, y_train, validation_split=0.1)

# Оценка модели
loss, accuracy = model.evaluate(X_test, y_test)
print(f"Точность модели: {accuracy}")


Trial 10 Complete [00h 09m 29s]
val_accuracy: 1.0

Best val_accuracy So Far: 1.0
Total elapsed time: 01h 30m 20s
1556/1556 [==============================] - 5s 3ms/step - loss: 3.5635e-04 - accuracy: 1.0000
Точность модели: 1.0


    Trial 10 Complete [00h 09m 29s]: Это указывает на то, что десятая попытка (или итерация) обучения модели была завершена за 9 минут 29 секунд.

    Best val_accuracy So Far: 1.0: Самая высокая достигнутая точность на валидационном наборе данных составила 1.0 или 100%. Это означает, что модель идеально классифицировала валидационные данные.

    Total elapsed time: 01h 30m 20s: Общее время, затраченное на все попытки обучения модели до этого момента, составило 1 час 30 минут 20 секунд.

    6221/6221 [==============================] - 25s 4ms/step - loss: 0.1171 - accuracy: 0.9660: Это результаты обучения на обучающем наборе данных. Здесь указано, что модель обработала все 6221 образца, затратив в среднем 4 миллисекунды на шаг. Средняя потеря (loss) составила 0.1171, а точность (accuracy) – 96.6%. Эти показатели говорят о том, что модель хорошо справляется с обучающими данными.

    1556/1556 [==============================] - 5s 3ms/step - loss: 3.5635e-04 - accuracy: 1.0000: Это результаты на валидационном наборе данных. Модель обработала все 1556 образцов, потратив в среднем 3 миллисекунды на шаг. Потеря (loss) составила всего 0.00035635, а точность (accuracy) достигла идеальных 100%.

    Точность модели: 1.0: Это подтверждение того, что модель достигла идеальной точности на валидационном наборе данных.

Такие результаты могут указывать на то, что модель отлично справляется с поставленной задачей. Однако стоит учесть, что точность 100% может также быть признаком переобучения, особенно если есть значительное расхождение между точностью на обучающем и валидационном наборах данных. В таком случае рекомендуется провести дополнительную проверку модели на тестовом наборе данных, который не использовался в процессе обучения, чтобы убедиться в ее способности обобщать и применять полученные знания к новым данным.